### ISSUE TITLE
Nice to have: Add optional identity parameters to LetsEncrypt TLS cert requests.

### ISSUE THREAD

**Issue by chris001** [virtualmin/virtualmin-gpl#388]
When Virtualmin makes a LetsEncrypt cert request, it appears to omit the optional `certbot` command line parameters to identify the Country (US), State (California), Locality (San Francisco), Organization (Github, Inc.), Common name (github.com), email of the server admin (used by letsencrypt to send you notification emails when cert is expiring soon or other issues with any of your other certs associated with this email address), etc., therefore these fields come back as blank on the cert details, which can be bad.

For some websites, it looks untrustworthy, or less than competent, to have a blank (anonymous) organization on the cert, no nothing, all blank.  What are you hiding.  Why are you hiding.  Or are you just lacking skill in requesting your certs and don't know how to provide these optional parameters so they get filled in.

Besides, some countries in the EU **require** websites who interact with the public, to identify the organization behind the website, so filling in these optional fields on the LetsEncrypt TLS certs could help clear up any doubts about regulatory compliance.

Therefore, for greater level of trust, the LetsEncrypt TLS cert requests should allow the server admin the option to auto fill in, or manually fill in, all the optional identifying info which LetsEncrypt `certbot` takes as optional parameters on the cert request.  Then, the cert will be all filled in, same as when you view details on a paid cert from the commercial cert vendors e.g. DigiCert, Comodo, etc.

**Comment by jcameron** [comment]
That's a good idea - I'll work on this, and update this ticket when it's done.

**Comment by jcameron** [comment]
So I started to look into this ...  but I couldn't find which command line flags to the `certbot` command allow the country, state or organization to be specified? Are there any docs on how to do this manually?

**Comment by chris001** [comment]
The admin email address (for important notifications, such as expiration, etc, from EFF/Let's Encrypt) is available on the command line with `-m EMAIL`.  I believe the identify info (Country, State, Locality, Organization, Common Name) is part of the CSR (cert signing request) which is a text file IIRC.  Docs: https://eff-certbot.readthedocs.io/en/stable/using.html#certbot-command-line-options

**Comment by jcameron** [comment]
Ah, so in order to set the Country/State/etc, Webmin would need to generate a CSR separately and then run `certbot` with the `--csr` flag. This would be a change from how it works currently, where the CSR is generated entirely by `certbot` .

**Comment by chris001** [comment]
Yes, it might need a two-phase process, one phase to generate the CSR, call `certbot` , get a cert back, then maybe another phase to call `certbot` to have it install the cert into the web server with whichever suitable browser plugin, if that's how the cert gets installed now? Kind of clumsy but there should be a way to accomplish with a CSR the same as the process which is lacking the CSR identity info.

**Comment by jcameron** [comment]
I'm actually surprised that certbot doesn't have a built-in way to set the country/etc in a cert.  I will look into doing it the two-step way, which shouldn't be too hard as Virtualmin already has support for generating a CSR when requesting an SSL cert from a traditional CA.

**Comment by chris001** [comment]
I'm surprised too., from [the docs](https://eff-certbot.readthedocs.io/en/stable/using.html#certbot-command-line-options), it looks like `--csr` works only with `certonly` .  Surely there must be a well known method to run it twice, getting `certbot` to install the cert on the second run.

**Comment by iliaross** [comment]
Reopened, as this was closed by an automated cleanup run.

⚠️ If the issue is resolved, please take a moment to close it to help keep the project organized.
